### PR TITLE
fix: unify code testing & step testing

### PR DESCRIPTION
**Pull request by abuaboud**
## What does this PR do?

Fixes # (issue)

### CODE REVIEW

File: packages/backend/src/app/pieces/pieces.controller.ts [hunk 1]
```diff
@@ -113,7 +113,10 @@ async function stats(): Promise<Record<string, PieceStats>> {
...
     for (const flowWithoutVersion of flows) {
         const flow = await flowService.getOneOrThrow({ id: flowWithoutVersion.id, projectId: flowWithoutVersion.projectId })
-        const trigger = flow?.version?.trigger
+        if(isNil(flow.version)) {
+            continue
+        }
+        const trigger = flow.version.trigger
         if (isNil(trigger)) {
             continue
         }
```
Comment (AbdulTheActivePiecer) [inline packages/backend/src/app/pieces/pieces.controller.ts:122]: These cases shouldn't exist and if they do you might want to throw an error instead to be knowledgeable of these cases.

File: packages/shared/src/lib/flows/step-run/dto/test-step-request.ts [hunk 2]
```diff
@@ -1,6 +1,15 @@
-import { Type } from "@sinclair/typebox";
+import { Static, Type } from "@sinclair/typebox";
 
 export const CreateStepRunRequestBody = Type.Object({
     flowVersionId: Type.String(),
     stepName: Type.String(),
 })
+
+export const CreateStepRunResponse = Type.Object({
```
Comment (AbdulTheActivePiecer) [inline packages/shared/src/lib/flows/step-run/dto/test-step-request.ts]: Why not rename it to stepRunResponse ?

File: packages/backend/src/app/helper/engine-helper.ts [hunk 3]
```diff
@@ -240,6 +248,28 @@ export const engineHelper = {
         }
     },
 
+    async executeCode(operation: ExecuteCodeOperation): Promise<EngineHelperResponse<EngineHelperCodeResult>> {
+        logger.debug(operation, '[EngineHelper#executeAction] operation')
+
+        const sandbox = await sandboxManager.obtainSandbox(`${apId()}`)
```
Comment (AbdulTheActivePiecer) [inline packages/backend/src/app/helper/engine-helper.ts]: Unnecessary string interpolation.

File: packages/backend/src/app/pieces/pieces.controller.ts (hunk 1 above)
Comment (abuaboud) [reply packages/backend/src/app/pieces/pieces.controller.ts:122]: It does exists, I will investigate soon why they are there.

**Pull request merged**